spec/puppet/resource_api_spec.rb

```diff
@@ -408,105 +408,308 @@ def set(_context, changes)
     before(:each) do
       stub_const('Puppet::Provider::Canonicalizer', Module.new)
       stub_const('Puppet::Provider::Canonicalizer::Canonicalizer', provider_class)
+      Puppet.settings[:strict] = :off
     end
 
-    it { expect { described_class.register_type(definition) }.not_to raise_error }
+    context 'with get values that will not be canonicalized' do
+      it { expect { described_class.register_type(definition) }.not_to raise_error }
 
-    describe 'the registered type' do
-      subject(:type) { Puppet::Type.type(:canonicalizer) }
+      describe 'the registered type' do
+        subject(:type) { Puppet::Type.type(:canonicalizer) }
 
-      before(:each) do
-        allow(type.my_provider).to receive(:get)
-          .with(kind_of(Puppet::ResourceApi::BaseContext))
-          .and_return([{ name: 'somename', test_string: 'canonfoo' },
-                       { name: 'other', test_string: 'canonbar' }])
-      end
+        before(:each) do
+          allow(type.my_provider).to receive(:get)
+            .with(kind_of(Puppet::ResourceApi::BaseContext))
+            .and_return([{ name: 'somename', test_string: 'canonfoo' },
+                         { name: 'other', test_string: 'canonbar' }])
```

Comment:
This is duplicating a big chunk of setup code with the new `with get values that will be canonicalized` section, without extracting the common parts. It would be easier to read if you do something like this:

```
before(:each) do
  allow(type.my_provider).to receive(:get)
    .with(kind_of(Puppet::ResourceApi::BaseContext))
    .and_return([{ name: 'somename', test_string: first_test_string },
                  { name: 'other', test_string: second_test_string }])
end

context 'with get values that will not be canonicalized' do
  let(:first_test_string) { "canonfoo" }
  let(:first_test_string) { "canonbar" }

  # ...
end

context 'with get values that will be canonicalized' do
  let(:first_test_string) { "foo" }
  let(:first_test_string) { "bar" }

  # ...
end
```